### DISCUSSION TITLE
MyST parser feature

### DISCUSSION THREAD

**Discussion by pypamart** [mkdocs/mkdocs#2899] [Ideas]
Hello Mkdocs teams,

Thank you very much for your great software!
It could be very nice to have the choice to select the MyST parser (https://myst-parser.readthedocs.io/en/latest/index.html).
Have you plan to implement this option?

Thank you

**Comment by peekxc** [comment]
Unfortunately, despite the docs suggesting otherwise, the current functionality of MyST is quite lacking in terms of actually embedding code-outputs.

This is easy to verify yourself, e.g. if you have a markdown file `myst.md`

```markdown 
---
kernelspec:
  name: python
  display_name: 'Python'
---

## Testing MyST parser 

'''{python}
print("hello")
'''

'''{code-cell} python
x = "hello"
y = "there"
phrase = f"{x}, {y}!"
print(phrase)
'''

'''{code-cell} python
from bokeh.plotting import figure, show
from bokeh.io import output_notebook
output_notebook()

p = figure(width=200, height=200)
p.scatter(x=list(range(10)), y=list(range(10)))
show(p)
'''
```

(NOTE: replace the `'''` with triple backticks, this is just to get the point across). 

And you run something like:

```bash
myst build myst.md --execute --md   
```

You end up with something kind of horrendous, an entire `_build` directory with all of the outputs, figures, and even log files stores in separate file: 
 
```
├── execute
│   └── e50b4874c116114469fc484c5c6203fb.json
├── exports
│   ├── files
│   │   ├── 018a72da10e6f9fc78c289eecda2be11.json
│   │   ├── 103119d56f23edde7c7e52d4ec33d6e3.txt
│   │   ├── 1df358db547460264a71f8440907de6d.json
│   │   ├── 2447ad3cf077d2afd229237521e47d02.html
│   │   ├── 5c8b54167f029b207ad9891c09f08ba6.json
│   │   └── f9c647f03d16c9895a6bd7eeb0ba227e.html
│   ├── myst.md
└── logs
    └── myst.build.json
```

The resulting `myst.md` unfortunately will not be known at all to `mkdocs`, 


```markdown 
---
kernelspec:
  name: python
  display_name: 'Python'
content_includes_title: true
parts: {}
...
---

## Testing MyST parser 

+++
'''{python}
print("hello")
'''

+++
'''{code-cell} python
x = "hello"
y = "there"
phrase = f"{x}, {y}!"
print(phrase)
'''

+++
'''{code-cell} python
from bokeh.plotting import figure, show
from bokeh.io import output_notebook
output_notebook()

p = figure(width=200, height=200)
p.scatter(x=list(range(10)), y=list(range(10)))
show(p)
'''
```

For w/e reason, the MyST team decided to keep all the outputs as separate files, w/o the option to keep everything inline, probably because they fear that would create something not too away from being just a full-fledged jupyter notebook (indeed, the .json file in the `execute` folder is 25Kb). One could probably write a plugin to incorporate those files transparently during the build process but it might not be very useful. 

It seems to be the only viable options for including embedding Jupyter-like code outputs into markdown files is [quarto](https://quarto.org/), which does have a (fairly rudimentary) [mkdocs plugin](https://jspaezp.github.io/mkquartodocs/), though admittedly this kind of slow / you lose a lot fo the speed and simplicity of mkdocs due to the internal use of pandoc. 

There's also this [plugin](https://github.com/danielfrg/mkdocs-jupyter) that has some support for jupytext, python files, and jupyter notebooks. This option seems ok for simple outputs, but I think won't have much in terms of configurability
